src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -757,13 +757,26 @@ export class BrowsingContextImpl {
       );
     }
 
-    const result = await this.#cdpTarget.cdpClient.sendCommand(
-      'Page.captureScreenshot',
-      {clip: {...rect, scale: 1.0}}
-    );
-    return {
-      data: result.data,
-    };
+    try {
+      await this.#cdpTarget.cdpClient.sendCommand(
+        'Emulation.setDefaultBackgroundColorOverride',
+        {
+          color: {r: 0, g: 0, b: 0, a: 0},
+        }
+      );
+      return await this.#cdpTarget.cdpClient.sendCommand(
+        'Page.captureScreenshot',
+        {clip: {...rect, scale: 1.0}}
+      );
+    } finally {
+      await this.#cdpTarget.cdpClient
+        .sendCommand('Emulation.setDefaultBackgroundColorOverride', {
+          color: undefined,
```

Comment:
What if the background was previously set to some custom color? Would this command reset it to the previous value?

Reply:
Yes

Reply:
Let me rephrase to double-check:

- Assume background color is r=0 g=255 b=0 (green)
- If I am a BiDi+ user who called `Emulation.setDefaultBackgroundColorOverride(r: 255, g: 0, b: 0, a: 0)` (override to red),
- Then called `capture screenshot`

Will the background color be reset back to red afterwards?

It seems to me it would be reset, but to green.

This is unexpected, as it's not in the contract of the capture screenshot method. Right?

Reply:
So the I would expect the color to be back to green assuming we have the following sequence of calls and CDP is working as expected:

1) setDefaultBackgroundColorOverride(green, sessionA) # overrides = (green)
2) setDefaultBackgroundColorOverride(red, sessionB) # overrides = (green, red)
3) setDefaultBackgroundColorOverride(white, sessionA) # overrides = (green, white)
4) setDefaultBackgroundColorOverride(undefined, sessionA) # overrides = (green)

That is how overrides in CDP should work, if it turns out that this override works differently we could see if we want to get that fixed in CDP. Also, this method is usually only called for screenshot taking with the white bg color.

Reply:
Looks like we would need to fix CDP to make this CDP API work properly across sessions https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/inspector_emulation_agent.cc;l=582;drc=58b8aff108cb11d3755f4cc18efa10f971be9682;bpv=1;bpt=1